ddl/reorg.go

```diff
@@ -444,15 +455,28 @@ func getValidCurrentVersion(store kv.Storage) (ver kv.Version, err error) {
 	return ver, nil
 }
 
-func getReorgInfo(d *ddlCtx, t *meta.Meta, job *model.Job, tbl table.Table) (*reorgInfo, error) {
+func getReorgInfo(d *ddlCtx, t *meta.Meta, job *model.Job, tbl table.Table, elements []*meta.Element) (*reorgInfo, error) {
 	var (
-		start kv.Handle
-		end   kv.Handle
-		pid   int64
-		info  reorgInfo
+		element *meta.Element
+		start   kv.Handle
+		end     kv.Handle
+		pid     int64
+		info    reorgInfo
 	)
 
 	if job.SnapshotVer == 0 {
+		// For the case of the old TiDB version(do not exist the element information) is upgraded to the new TiDB version.
+		// Third step, we need to remove the element information to make sure we can save the reorganized information to storage.
+		failpoint.Inject("MockGetIndexRecordErr", func(val failpoint.Value) {
+			if val.(string) == "addIdxNotOwnerErr" && atomic.CompareAndSwapUint32(&mockNotOwnerErrOnce, 3, 4) {
+				if err := t.RemoveReorgElement(job); err != nil {
+					failpoint.Return(nil, errors.Trace(err))
+				}
+				info.first = true
+				failpoint.Return(&info, nil)
```

Comment:
seems we don't need return here, we can directly walkthrough to the below code.

Reply:
`GetDDLReorgHandle` is in `else` branch, this code is in `if` branch, PTAL

Reply:
If remove the return, `UpdateDDLReorgHandle` will be called, then we do `RemoveReorgElement` is useless.

Reply:
got it, pls added this in L475
// make `RemoveReorgElement` commit success in a single round kv transation